plugins/pencilblue/controllers/actions/admin/sites/localization.js

```diff
@@ -0,0 +1,268 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+var path = require('path');
+var fs = require('fs');
+
+module.exports = function(pb) {
+
+    //pb dependencies
+    var util = pb.util;
+
+    /**
+     * Saves the site's Localization settings
+     */
+    function Localization(){}
+    util.inherits(Localization, pb.BaseAdminController);
+
+    Localization.prototype.render = function(cb) {
+        var self = this;
+
+        this.getJSONPostParams(function(err, post) {
+            if(util.isError(err)) {
+                return cb({
+                    code: 500,
+                    content: pb.BaseController.apiResponse(pb.BaseController.API_FAILURE, self.ls.get('ERROR_SAVING'), err)
+                });
+            }
+
+            if(pb.config.localization && pb.config.localization.db){
+                return self.saveLocalesToDatabase(post, cb);
+            }
+
+            Localization.prototype.saveLocalesToFile(post, cb);
+        });
+
+    };
+
+    Localization.prototype.saveLocalesToFile = function(post, cb){
+        var self = this;
+        self.query = post;
+        self.getLocalesFromFile(function(err, pluginsJsonFileObj) {
+            if (err)
+                throw err;
+            if (pluginsJsonFileObj) {
+                var obj = {};
+                post.translations.forEach(function (element) {
+                    obj[element.key] = element.value;
+                });
+                pluginsJsonFileObj[post.site] = obj;
+                try {
+                    pluginsJsonFileObj = JSON.stringify(pluginsJsonFileObj);
+                } catch (e) {
+                    pb.log.error(e);
+                    return cb({
+                        code: 500,
+                        content: pb.BaseController.apiResponse(pb.BaseController.API_FAILURE, self.ls.get('ERROR_SAVING'), e)
+                    });
+                }
+                self.sendLocalesToFile(pluginsJsonFileObj, cb);
+            }
+        });
+    };
+
+    Localization.prototype.saveLocalesToDatabase = function(post, cb){
```

Comment:
There is a lot of heavy lifting done by this controller.  abstracting it out to a service will make the controller more readable and reduce overall complexity.

The service should inherit from BaseObjectService.  It will eliminate most of the logic in this function